### PR TITLE
[test] add optional csv output to view buffer test cases in qgis

### PR DESCRIPTION
**Pull request by barendgehrels**

### CODE REVIEW

File: test/algorithms/buffer/test_buffer_csv.hpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        stream << std::setprecision(16)
+            << "id;role;wkt" << std::endl
+            << "1;input;" << bg::wkt(input) << std::endl
+            << "2;output;" << bg::wkt(buffer) << std::endl;
+    }
+
+private:
+
+    template <typename Turns>
+    inline void write_turns(Turns const &turns)
```
Comment (vissarion) [inline test/algorithms/buffer/test_buffer_csv.hpp:63]: We write it `Turns const& turns`

File: test/algorithms/buffer/test_buffer_csv.hpp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+class buffer_visitor_csv
+{
+public:
+    buffer_visitor_csv(const std::string &path)
+        : m_path(path)
+    {
+    }
+
+    template <typename PieceCollection>
+    inline void apply(PieceCollection const &collection, int phase)
```
Comment (vissarion) [inline test/algorithms/buffer/test_buffer_csv.hpp:33]: We write it `PieceCollection const& collection`

File: test/algorithms/buffer/test_buffer_csv.hpp [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                   << bg::operation_char(turn.operations[0].operation) << "/" << bg::operation_char(turn.operations[1].operation) << ";"
+                   << turn.cluster_id << ";"
+                   << turn.is_turn_traversable << ";"
+                   << turn.blocked() << ";"
+                   << std::endl;
+        }
+    }
+
+    template <typename Pieces, typename OffsettedRings>
+    inline void write_pieces(Pieces const &pieces, OffsettedRings const &offsetted_rings)
```
Comment (vissarion) [inline test/algorithms/buffer/test_buffer_csv.hpp:87]: as above we use `const& varaible_name`

File: test/algorithms/buffer/test_buffer_csv.hpp [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        if (! stream.good())
+        {
+            return;
+        }
+
+        std::size_t id = 1;
+        stream << std::setprecision(16) << "id;wkt;index;start;end;type;segment_indices" << std::endl;
+        for (const auto &piece : pieces)
+        {
+            bg::segment_identifier seg_id = piece.first_seg_id;
```
Comment (vissarion) [inline test/algorithms/buffer/test_buffer_csv.hpp:99]: should this be `const` ?

**Review by vissarion — APPROVED** [review]
LGTM, thanks!

File: test/algorithms/buffer/debug_buffer_info.hpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+// Unit Test Helper
+
+// Copyright (c) 2022 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+
+#ifndef BOOST_GEOMETRY_TEST_BUFFER_DEBUG_BUFFER_INFO_HPP
```
[the remark this answers is not in the document]
Comment (barendgehrels) [reply test/algorithms/buffer/debug_buffer_info.hpp:11]: Sure, thanks. :heavy_check_mark:

File: test/algorithms/buffer/test_buffer_csv.hpp (hunk 1 above)
Comment (barendgehrels) [reply test/algorithms/buffer/test_buffer_csv.hpp:63]: :heavy_check_mark:

File: test/algorithms/buffer/test_buffer_csv.hpp (hunk 4 above)
Comment (barendgehrels) [reply test/algorithms/buffer/test_buffer_csv.hpp:99]: Sure, thanks. Updated it and your other comments.

**Pull request merged**